lib/fluent/plugin/buffer/chunk.rb

```diff
@@ -141,19 +144,40 @@ def purge
           self
         end
 
-        def read
+        def read(**kwargs)
           raise NotImplementedError, "Implement this method in child class"
         end
 
         def open(&block)
           raise NotImplementedError, "Implement this method in child class"
         end
 
-        def write_to(io)
+        def write_to(io, **kwargs)
           open do |i|
             IO.copy_stream(i, io)
           end
         end
+
+        module Decompressable
+          include Fluent::Plugin::Compressable
+
+          def read(**kwargs)
+            return super if kwargs[:compress] == :gzip
+
+            # avoid creating duplicated IO
+            if @chunk.is_a?(IO)
+              # reset io(@chunk) to read
+              @chunk.seek(0, IO::SEEK_SET)
+              decompress('', io: @chunk) # not to call IO#read
```

Comment:
Specifying blank string for unused argument looks bad API.
Is there a way to use default value for that parameter? Or any other solution? (At least, `nil` is better for unused parameter value.)